packages/wattles/lib/src/schemas/schema_invocation.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:equatable/equatable.dart';
+
+/// {@template schema_invocation}
+///
+/// {@endtemplate}
+class SchemaInvocation extends Equatable implements Exception {
+  /// {@macro schema_invocation}
+  ///
+  /// Tests the given [func] and if it throws an [SchemaInvocation] it returns
+  /// the [SchemaInvocation] otherwise it throws an [UnimplementedError]
+  /// stating that the given [func] has not implemented a [SchemaInvocation].
+  factory SchemaInvocation(
+    Function func, {
+    List<dynamic> arguments = const [],
+  }) {
+    try {
+      Function.apply(func, arguments);
+    } on SchemaInvocation catch (err) {
+      return err;
```

Comment:
![](https://media.giphy.com/media/nTfdeBvfgzV26zjoFP/giphy.gif)

Reply:
It is all about hacks <3